baseapp/baseapp.go

```diff
@@ -911,6 +911,12 @@ func (app *BaseApp) runTx(mode execMode, txBytes []byte) (gInfo sdk.GasInfo, res
 		gasWanted = ctx.GasMeter().Limit()
 
 		if err != nil {
+			if mode == execModeReCheck {
+				// if the ante handler fails on recheck, we want to remove the tx from the mempool
+				if err := app.mempool.Remove(tx); err != nil {
```

Comment:
Do we need to check if the tx exists in app-side mempl before removal?

Reply:
Will add thanks

Reply:
the weird part with checking if it exists is that we would need to iterate the entire mempool to find it since we cant assume the underlying data strucuture. Im not sure if its worth the trade off. The assmuption that txs can not be removed from the mempool without comet removign it first or the size of the mempool would make me think its safe to blidnly remove. Do you think this is a safe assumption/what would you expect as a user?